garage/envs/util.py

```diff
@@ -1,7 +1,7 @@
 import gym
```

Comment:
we can't delete this file entirely?

Reply:
Not yet no

Reply:
What's still in use?

Reply:
```
(garage) garage (replace_distributions) $ ag "from garage.envs.util import "
contrib/bichengcao/examples/trpo_gym_Acrobot-v1.py
5:from garage.envs.util import horizon, spec

contrib/bichengcao/examples/trpo_gym_CartPole-v0.py
5:from garage.envs.util import horizon, spec

contrib/bichengcao/examples/trpo_gym_Pendulum-v0.py
5:from garage.envs.util import horizon, spec

contrib/bichengcao/examples/trpo_gym_CartPole-v1.py
5:from garage.envs.util import horizon, spec

contrib/bichengcao/examples/trpo_gym_MountainCar-v0.py
6:from garage.envs.util import horizon, spec

contrib/ros/envs/sawyer/launchers/trpo_real_sawyer_push.py
6:from garage.envs.util import spec

contrib/ros/envs/sawyer/launchers/trpo_real_sawyer_pnp.py
6:from garage.envs.util import spec

contrib/ros/envs/sawyer/launchers/trpo_sim_sawyer_pnp.py
6:from garage.envs.util import spec

contrib/ros/envs/sawyer/launchers/trpo_real_sawyer_reacher.py
8:from garage.envs.util import spec

contrib/ros/envs/sawyer/launchers/trpo_sim_sawyer_push.py
6:from garage.envs.util import spec

contrib/ros/envs/sawyer/launchers/trpo_sim_sawyer_reacher.py
11:from garage.envs.util import spec

examples/tf/cluster_gym_mujoco_demo.py
7:from garage.envs.util import spec

examples/tf/trpo_gym_tf_cartpole.py
4:from garage.envs.util import spec

garage/envs/box2d/box2d_env.py
12:from garage.envs.util import bounds, flat_dim

garage/envs/box2d/car_parking_env.py
8:from garage.envs.util import flat_dim

garage/envs/mujoco/gather/gather_env.py
24:from garage.envs.util import flat_dim

garage/envs/mujoco/hill/terrain.py
14:from garage.envs.util import flat_dim

garage/envs/mujoco/maze/maze_env.py
17:from garage.envs.util import flat_dim

garage/envs/mujoco/mujoco_env.py
16:from garage.envs.util import bounds

garage/envs/noisy_env.py
6:from garage.envs.util import flat_dim

garage/envs/normalized_env.py
5:from garage.envs.util import flat_dim, flatten, unflatten

garage/envs/occlusion_env.py
9:from garage.envs.util import flat_dim

garage/tf/algos/ddpg.py
19:from garage.envs.util import bounds
20:from garage.envs.util import flat_dim

garage/tf/exploration_strategies/ou_strategy.py
11:from garage.envs.util import flat_dim

garage/tf/policies/continuous_mlp_policy.py
11:from garage.envs.util import flat_dim

garage/tf/q_functions/continuous_mlp_q_function.py
4:from garage.envs.util import flat_dim

garage/theano/policies/categorical_conv_policy.py
6:from garage.envs.util import flatten, flatten_n, weighted_sample

tests/envs/test_sawyer_envs.py
12:from garage.envs.util import spec

tests/test_normalized_gym.py
7:from garage.envs.util import flat_dim, flatten
```

Reply:
I thought this PR guarantees that everything in garage receives a garage.spaces instance.

If that's true, can't those all be replaced by some_space.flatten(foo), some_space.spec, some_space.flat_dim,  some_space.bounds, etc. thanks to this PR?

Reply:
Furthermore, we will need to refactor a lot of Envs to inherit from GarageEnv (e.g. PickAndPlaceEnv, ReacherEnv, PushEnv)

Reply:
We can touch whatever we want, we just don't make development a priority in contrib. There's no point in making more technical debt when it's easy to avoid (e.g. I could implement the commit using `sed`).

Why not have ReacherEnv, etc. inherit from gym.Env as now? I can't see a use case for inheriting from GarageEnv/TfEnv/TheanoEnv. All environments are still supposed to be gym.Envs.